nequip/train/trainer.py

```diff
@@ -458,6 +469,15 @@ def from_dict(cls, dictionary, append: Optional[bool] = None):
 
         d = deepcopy(dictionary)
 
+        for code in [e3nn, nequip, torch, torch_geometric]:
+            version = d.get(f"{code.__name__}_version", None)
+            if version is not None and version != code.__version__:
+                raise NotImplementedError(
+                    "Parsing model from a different library version is not supported."
+                    f"current {code.__name__} verion: {code.__version} "
+                    f"vs  version nedded {version}"
+                )
```

Comment:
Lets do a warning instead of an error

For upgrades, very well may be backward compat — just want people to be aware of whats happening

Reply:
sure. just changed.